x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/KqlAstBuilder.java

```diff
@@ -8,22 +8,222 @@
+            } else {
+                builder.must(ParserUtils.typedParsing(this, subQueryCtx, QueryBuilder.class));
+            }
+        }
+
+        return rewriteConjunctionQuery(builder);
+    }
+
+    public QueryBuilder visitOrBooleanQuery(KqlBaseParser.BooleanQueryContext ctx) {
+        BoolQueryBuilder builder = QueryBuilders.boolQuery().minimumShouldMatch(1);
+
+        for (ParserRuleContext subQueryCtx : ctx.query()) {
+            if (subQueryCtx instanceof KqlBaseParser.BooleanQueryContext booleanSubQueryCtx && isOrQuery(booleanSubQueryCtx)) {
+                ParserUtils.typedParsing(this, subQueryCtx, BoolQueryBuilder.class).should().forEach(builder::should);
+            } else {
+                builder.should(ParserUtils.typedParsing(this, subQueryCtx, QueryBuilder.class));
+            }
+        }
+
+        return rewriteDisjunctionQuery(builder);
+    }
+
+    @Override
+    public QueryBuilder visitNotQuery(KqlBaseParser.NotQueryContext ctx) {
+        return QueryBuilders.boolQuery().mustNot(ParserUtils.typedParsing(this, ctx.simpleQuery(), QueryBuilder.class));
+    }
+
+    @Override
+    public QueryBuilder visitParenthesizedQuery(KqlBaseParser.ParenthesizedQueryContext ctx) {
+        return ParserUtils.typedParsing(this, ctx.query(), QueryBuilder.class);
+    }
+
+    @Override
+    public QueryBuilder visitNestedQuery(KqlBaseParser.NestedQueryContext ctx) {
+        // TODO: implementation
+        return new MatchNoneQueryBuilder();
+    }
+
+    @Override
+    public QueryBuilder visitMatchAllQuery(KqlBaseParser.MatchAllQueryContext ctx) {
+        return new MatchAllQueryBuilder();
+    }
+
+    @Override
+    public QueryBuilder visitExistsQuery(KqlBaseParser.ExistsQueryContext ctx) {
+        assert ctx.fieldName() != null; // Should not happen since the grammar does not allow the fieldname to be null.
+
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery().minimumShouldMatch(1);
+        withFields(ctx.fieldName(), (fieldName, mappedFieldType) -> {
+            if (isRuntimeField(mappedFieldType) == false) {
+                boolQueryBuilder.should(QueryBuilders.existsQuery(fieldName));
+            }
+        });
+
+        return rewriteDisjunctionQuery(boolQueryBuilder);
+    }
+
+    @Override
+    public QueryBuilder visitRangeQuery(KqlBaseParser.RangeQueryContext ctx) {
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery().minimumShouldMatch(1);
+
+        String queryText = ParserUtils.extractText(ctx.rangeQueryValue());
+        BiFunction<RangeQueryBuilder, String, RangeQueryBuilder> rangeOperation = rangeOperation(ctx.operator);
+
+        withFields(ctx.fieldName(), (fieldName, mappedFieldType) -> {
+            RangeQueryBuilder rangeQuery = rangeOperation.apply(QueryBuilders.rangeQuery(fieldName), queryText);
+            // TODO: add timezone for date fields
```

Comment:
⚠️ Will be exposed as a parameter of the `kql` query and done when introducing the query